src/django_components/component.py

```diff
@@ -41,24 +24,29 @@
     register,
     registry,
 )
+from django_components.component_registry import registry as component_registry
 from django_components.logger import logger
-from django_components.templatetags.component_tags import (
-    FILLED_SLOTS_CONTENT_CONTEXT_KEY,
+from django_components.middleware import is_dependency_middleware_active
+from django_components.slots import (
     DefaultFillContent,
-    FillContent,
-    FilledSlotsContext,
-    IfSlotFilledConditionBranchNode,
+    ImplicitFillNode,
     NamedFillContent,
+    NamedFillNode,
     SlotName,
-    SlotNode,
+    render_component_template_with_slots,
 )
 from django_components.utils import search
 
-_T = TypeVar("_T")
+RENDERED_COMMENT_TEMPLATE = "<!-- _RENDERED {name} -->"
 
 
 class SimplifiedInterfaceMediaDefiningClass(MediaDefiningClass):
     def __new__(mcs, name: str, bases: Tuple[Type, ...], attrs: Dict[str, Any]) -> Type:
+        # NOTE: Skip template/media file resolution when then Component class ITSELF
+        # is being created.
+        if "__module__" in attrs and attrs["__module__"] == "django_components.component":
```

Comment:
Genuine change - had to add this check to avoid circular imports. This check ensures that the template/media file search logic is NOT run when the `Component` class is defined below in this file.

This makes sense - we want to resolve media files only for SUBCLASSES of `Component`, not `Component` itself.